### DISCUSSION TITLE
Support to spatie/laravel-model-states

### DISCUSSION THREAD

**Discussion by ricventu** [filamentphp/filament#5326] [Ideas]
Hi,
What about to make Forms\Components\Select to handle [spatie/laravel-model-states](https://spatie.be/docs/laravel-model-states/v2/01-introduction)?
Maybe like: `Forms\Components\Select::make('state')->options(MyModelState::class)`

When selected option changes, make `state->transitionTo( MyOption::class)`

**Comment by maartenpaauw** [comment]
I’ve released a plugin which fits your needs! See: https://filamentphp.com/plugins/maartenpaauw-model-states
